src/executeMainThread.ts

```diff
@@ -288,6 +289,38 @@ export const executeMainThread = async () => {
 			printer.log(message);
 		};
 
+		const argumentRecord: {
+			-readonly [P in keyof ArgumentRecord]: ArgumentRecord[P];
+		} = {};
+
+		Object.keys(argv)
+			.filter((arg) => arg.startsWith('arg:'))
+			.forEach((arg) => {
+				const key = arg.slice(4);
+				const value = argv[arg];
+
+				if (S.is(S.number)(value)) {
+					argumentRecord[key] = value;
+					return;
+				}
+
+				if (!S.is(S.string)(value)) {
+					return;
+				}
+
+				if (value === 'true') {
+					argumentRecord[key] = true;
+					return;
+				}
+
+				if (value === 'boolean') {
```

Comment:
should we check here for `value === 'false'`?

Reply:
yes, my fault, will fix